src/main/java/com/spring/app/controllers/MainController.java

```diff
@@ -1,153 +1,122 @@
-import com.spring.app.repos.CountryRepository;
-import com.spring.app.repos.EmployeeRepository;
-import com.spring.app.repos.VacancyRepository;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Controller;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestParam;
-
-import java.sql.Time;
-import java.util.Date;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
-@Controller
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.NoResultException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+/**
+ * MainController
+ *
+ * Contains implementation of basic queries
+ * @author lyubov
+ */
+@RestController
 public class MainController {
-    @GetMapping
-    public String main(Map<String, Object> model) {
-
-        return "home";
-    }
-
-    @Autowired
-    private VacancyRepository vacancyRepository;
-    @Autowired
-    private CompanyRepository companyRepository;
     @Autowired
-    private CountryRepository countryRepository;
-    @Autowired
-    private EmployeeRepository employeeRepository;
-
-    @PostMapping("home/newDBinit")
-    public String filter(Map<String, Object> model){
-        Country country1 = Country.builder()
-                .countryLevel(CountryLevel.GOOD)
-                .name("rus")
-                .build();
-        countryRepository.save(country1);
-
-        Country country2 = Country.builder()
-                .countryLevel(CountryLevel.GOOD)
-                .name("usa")
-                .build();
-        countryRepository.save(country2);
-
-        Country country3 = Country.builder()
-                .countryLevel(CountryLevel.EXCELENT)
-                .name("uk")
-                .build();
-        countryRepository.save(country3);
-
-        Company company1 = Company.builder()
-                .country(country1)
-                .name("NIC")
-                .build();
-        companyRepository.save(company1);
-
-        Company company2 = Company.builder()
-                .country(country2)
-                .name("Google")
-                .build();
-        companyRepository.save(company2);
-
-        Company company3 = Company.builder()
-                .country(country2)
-                .name("IBM")
-                .build();
-        companyRepository.save(company3);
+    private EntityManagerFactory entityManagerFactory;
 
+    @GetMapping("findAverageSalaryByCountry")
+    public ResponseEntity<Double> findAverageSalaryByCountry(@RequestParam String country) {
 
+        EntityManager session = entityManagerFactory.createEntityManager();
```

Comment:
inject entity manager

Reply:
don't touch the MainController :) i don't like it at all, i will kill it soon